agent/command/generate.go

```diff
@@ -120,11 +120,10 @@ func (c *generateTask) Execute(ctx context.Context, comm client.Communicator, lo
 				return false, generateErr
 			}
 			if generateErr != nil {
-				// if the error isn't related to saving the generated task, log it but still retry in case of race condition
 				if !strings.Contains(generateErr.Error(), evergreen.SaveGenerateTasksError) {
-					logger.Task().Infof("Problem polling for generate tasks job, retrying (%s)", generateErr.Error())
+					logger.Task().Infof("Problem polling for generate tasks job", generateErr.Error())
```

Comment:
Now that we won't potentially poll forever when something is misconfigured we shouldn't need this special casing to make sure we tell the user what happened (or this SaveGenerateTasksError at all), it should log it at the end (you could test this by making a generate task with an invalid config and see if it logs)

Reply:
Sounds good, did this and tested in staging to confirm this log always hits on error.

Reply:
Well actually what i meant was -- do we need to have this log here or do we already log the error returned from the loop somewhere? (I would think we hit https://github.com/evergreen-ci/evergreen/blob/68c43797d61a0c21c338dfc7f08352b42a5ddff0/agent/command.go#L131 already so we're duplicating logging then)

Reply:
Ah yes - that line is duplicative. Removed.